newroutes/__tests__/Navigation.spec.js

```diff
@@ -0,0 +1,128 @@
+const express = require("express")
+const _ = require("lodash")
+const request = require("supertest")
+
+const { errorHandler } = require("@middleware/errorHandler")
+const { attachReadRouteHandlerWrapper } = require("@middleware/routeHandler")
+
+const { navigationContent, navigationSha } = require("@fixtures/navigation")
+
+const { NavigationRouter } = require("../navigation")
+
+describe("Navigation Router", () => {
+  const mockNavigationYmlService = {
+    read: jest.fn(),
+    update: jest.fn(),
+  }
+
+  const router = new NavigationRouter({
+    navigationYmlService: mockNavigationYmlService,
+  })
+
+  const app = express()
+  app.use(express.json({ limit: "7mb" }))
+  app.use(express.urlencoded({ extended: false }))
+
+  // We can use read route handler here because we don't need to lock the repo
+  app.get(
+    "/:siteName/navigation",
+    attachReadRouteHandlerWrapper(router.readNavigation)
+  )
+  app.post(
+    "/:siteName/navigation",
+    attachReadRouteHandlerWrapper(router.updateNavigation)
+  )
+  app.use(errorHandler)
+
+  const siteName = "test-site"
+  const accessToken = undefined // Can't set request fields - will always be undefined
+
+  const reqDetails = { siteName, accessToken }
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe("readNavigation", () => {
+    const expectedResponse = {
+      sha: navigationSha,
+      content: navigationContent,
+    }
+    mockNavigationYmlService.read.mockResolvedValue(expectedResponse)
+
+    it("retrieves navigation details", async () => {
+      const resp = await request(app).get(`/${siteName}/navigation`).expect(200)
+
+      expect(resp.body).toStrictEqual(expectedResponse)
+      expect(mockNavigationYmlService.read).toHaveBeenCalledWith(reqDetails)
+    })
+  })
+
+  describe("updateNavigation", () => {
+    it("rejects requests with invalid body", async () => {
+      await request(app).post(`/${siteName}/navigation`).send({}).expect(400)
+    })
+
+    it("rejects requests with incomplete body", async () => {
+      const reqBody = {
+        content: navigationContent,
+      }
+
+      await request(app)
+        .post(`/${siteName}/navigation`)
+        .send(reqBody)
+        .expect(400)
+    })
+
+    it("rejects requests with incorrect body", async () => {
+      // Multiple types of menu types
+      const incorrectDetails = _.cloneDeep(navigationContent)
+      incorrectDetails.links[0].resource_room = true
```

Comment:
sorry, could you help me to understand this? the validator allows **only** true, which seems to imply that this should be valid

Reply:
This test was for checking that only one of `url`/`collection`/`resource_room`/`url` + `sublinks` exist - but this item has both `url` and `resource_room`